dev-tools/oxlog/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+                };
+                // Add the path to the current logs for the zone
+                let mut dir = zones_path.clone();
+                dir.push(zone);
+                dir.push("root/var/svc/log");
+                let mut paths =
+                    Paths { primary: dir, debug: vec![], extra: vec![] };
+
+                // Add the path to the extra logs for the zone
+                if zone.starts_with("oxz_cockroachdb") {
+                    let mut dir = zones_path.clone();
+                    dir.push(zone);
+                    dir.push("root/data/logs");
+                    paths.extra.push(("cockroachdb", dir));
+                }
+
+                zones.insert(zone.to_string(), paths);
+            }
+        }
+
+        // Find the directories containing the debug log files
+        for uuid in &pools.external {
+            let zones_path: Utf8PathBuf =
+                ["/pool/ext", &uuid.to_string(), "crypt/debug"]
+                    .iter()
+                    .collect();
+            // Find the zones on the given pool
+            let Ok(entries) = read_dir(zones_path.as_path()) else {
+                continue;
+            };
+            for entry in entries {
+                let Ok(zone_entry) = entry else {
+                    continue;
+                };
+                let zone = zone_entry.file_name();
+                let Some(zone) = zone.to_str() else {
+                    // not utf8
+                    continue;
+                };
+                let mut dir = zones_path.clone();
+                dir.push(zone);
+
+                // We only add debug paths if the zones have primary paths
+                if let Some(paths) = zones.get_mut(zone) {
+                    paths.debug.push(dir);
+                }
+            }
+        }
+
+        Ok(Zones { zones })
+    }
+
+    /// Return log files organized by service name
+    pub fn zone_logs(
+        &self,
+        zone: &str,
+        filter: Filter,
+    ) -> BTreeMap<ServiceName, SvcLogs> {
+        let mut output = BTreeMap::new();
+        let Some(paths) = self.zones.get(zone) else {
+            return BTreeMap::new();
+        };
+        load_svc_logs(paths.primary.clone(), &mut output);
+
+        if filter.archived {
+            for dir in paths.debug.clone() {
+                load_svc_logs(dir, &mut output);
+            }
+        }
+        if filter.extra {
+            for (svc_name, dir) in paths.extra.clone() {
+                load_extra_logs(dir, svc_name, &mut output);
+            }
+        }
+        output
+    }
+}
+
+const OX_SMF_PREFIXES: [&str; 2] = ["oxide-", "system-illumos-"];
+
+/// Return true if the provided file name appears to be a valid log file for an
+/// Oxide-managed SMF service.
+///
+/// Note that this operates on the _file name_. Any leading path components will
+/// cause this check to return `false`.
+pub fn is_oxide_smf_log_file(filename: impl AsRef<str>) -> bool {
```

Comment:
@bnaecker I copied this over from `illumos_utils`. If you want I can go ahead and update the `zone-bundle` code to use the method here instead, and remove it from `illumos_utils`.  I can do that in this PR or a follow up. 

It also looks like (according to grep at least) that we aren't using `illumos_utils::is_oxide_smf_service`. Should I just remove it altogether? I can leave it in `illumos-utils` but it will seem like a hanging chad after removing `is_oxide_smf_log_file`.

Reply:
Nevermind. I misread the comment above. `illumos_utils::running_zone` now calls into the `oxlog` method.